Regional PD documentation

verult opened the pull request:
/cc @msau42 @pmallory

verult commented:
/assign @chenopis @saad-ali

msau42 commented on content/en/docs/concepts/storage/volumes.md:
```diff
@@ -338,6 +338,8 @@ simultaneous writers allowed.
 Using a PD on a Pod controlled by a ReplicationController will fail unless
 the PD is read-only or the replica count is 0 or 1.
 
+Support for [Regional Persistent Disks](https://cloud.google.com/compute/docs/disks/#repds) was introduced to Kubernetes v1.8 as alpha and moved to beta in Kubernetes v1.10. This feature allows the creation of Persistent Disks that are available in two zones within the same region. In order to use this feature, the volume must be provisioned as a PersistentVolume; referencing the volume directly  ...
```
You can use this tag to denote a beta feature:
```
{{< feature-state for_k8s_version="v1.10" state="beta" >}}
```

msau42 replied:
I don't think we need to mention 1.8 alpha because the storageclass parameters were also different.

msau42 commented on content/en/docs/concepts/storage/volumes.md:
```diff
@@ -368,6 +370,33 @@ spec:
+kind: PersistentVolume
+metadata:
+  name: test-volume
+  annotations:
```
label?

verult commented:
Addressed comments, PTAL

msau42 commented:
/lgtm

chenopis commented on content/en/docs/concepts/storage/volumes.md:
````diff
@@ -368,6 +368,38 @@ spec:
+```
+Example PersistentVolume spec:
+
+```yaml
````
Travis is failing when testing this YAML example. Double-check the syntax.
LGTM otherwise.

verult commented:
Moved regional PD manual provisioning instructions to the Persistent Volumes doc page because currently only examples of Pod specs are allowed in the Volumes page.

This is the first detailed example of a specific volume plugin in the Persistent Volumes page. Thoughts?

Will squash before merging.

msau42 commented on content/en/docs/concepts/storage/persistent-volumes.md:
```diff
@@ -581,6 +554,58 @@ Volume binding matrix for statically provisioned volumes:
+
+Raw Block Support exists for these plugins only.
+
+### Manually provisioning a GCE Regional Persistent Disk
```
Yeah I'm not sure about putting it here.  I think it is fine to put it in the Volumes page (I put local PV stuff there)

chenopis replied:
Problem was that the Travis test wouldn't allow it in the Volumes page.

msau42 replied:
How is it allowing local PV to be in the volumes page?

verult replied:
Was it enabled recently in Travis?

chenopis replied:
Not sure why the local PV example is not triggering a Travis error. The current version of [examples_test.go](https://github.com/kubernetes/website/blob/master/test/examples_test.go#L628) does not include the `PersistentVolume` kind for `/content/en/docs/concepts/storage/volumes.md`, just `Pod`.

verult replied:
Turns out it's because there's a space in the code block declaration: ` ``` yaml`

verult commented:
Reverted back to original doc layout. Also updated the validation code to allow PersistentVolumes.

/cc @tengqm 
I changed the validation logic as well, please review. Thanks!

msau42 commented:
storageclass changes lgtm

verult commented:
@msau42 is `volumeMode` necessary in the local volume example?

msau42 commented:
volumeMode is required to document the raw block feature, which is alpha.

chenopis commented:
/lgtm
/approve
